bluesky/utils.py

```diff
@@ -1149,7 +1155,28 @@ def apply_to_dict_recursively(d, f):
     return d
 
 
-class ProgressBar:
+class ProgressBar(abc.ABC):
+    def update(
+            self,
+            pos: Any,
+            *,
+            name: str = None,
+            current: Any = None,
+            initial: Any = None,
+            target: Any = None,
+            unit: str = "units",
+            precision: Any = None,
+            fraction: Any = None,
+            time_elapsed: float = None,
+            time_remaining: float = None,
+    ):
+        ...
+
+    def clear(self):
+        ...
+
+
+class AsciiProgressBar(ProgressBar):
```

Comment:
This is definitely a better name, but we should also retain `ProgressBar` as a backward-compatible alias to `AsciiProgressBar`, at least for awhile. A lot of code will need to be updated. Maybe the ABC could be called `ProgressBarBase` or something.

Reply:
Actually, to the extent that this is _sometimes_ handled by beamline scientists, would `TextProgressBar` or `TerminalProgressBar` be approximately as descriptive and a little more accessible?

Reply:
I think TerminalProgressBar is more descriptive, renamed. 
How do scientists use this class? My assumption was that they would only touch ProgressBarManager